content/_index.md

```diff
@@ -2,9 +2,39 @@
 title: "🍲 Based Cooking 🍳"
 ---
 
-## Categories
-
-{{< tagcloud >}}
+## What do you want to cook?
+
+<div class="search">
+  <input type="text" id="search" placeholder="Search...">
+  <button class="clear-search">
+    <svg xmlns="http://www.w3.org/2000/svg" class="ionicon" viewBox="0 0 512 512"><title>Backspace</title><path d="M135.19 390.14a28.79 28.79 0 0021.68 9.86h246.26A29 29 0 00432 371.13V140.87A29 29 0 00403.13 112H156.87a28.84 28.84 0 00-21.67 9.84v0L46.33 256l88.86 134.11z" fill="none" stroke="currentColor" stroke-linejoin="round" stroke-width="32"></path><path fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="32" d="M336.67 192.33L206.66 322.34M336.67 322.34L206.66 192.33M336.67 192.33L206.66 322.34M336.67 322.34L206.66 192.33"></path></svg>
+  </button>
+</div>
+
+<script>
+document.addEventListener('DOMContentLoaded', () => {
+  const rec = document.querySelectorAll('#artlist li')
+  const search = document.querySelector('#search')
+  const clearSearch = document.querySelector('.clear-search')
+
+  search.addEventListener('input', e => {
+    // grab search input value
+    const searchText = e.target.value.toLowerCase()
+    // for each recipe hide all but matched
+    rec.forEach(el => {
+      const recipeName = el.innerText.toLowerCase()
+      const isMatch = recipeName.includes(searchText)
+
+      el.hidden = !isMatch
+    })
+  })
```

Comment:
```suggestion
      if (hasFilter && isMatch) {
        matchCount++;
      }
    })

    ul.classList.toggle('is-filtered', matchCount > 0);
  })
```

Reply:
Tracks how many times a match was found, and if at least 1 match was found with a search, a class is applied to the `ul`.

Reply:
What's the reason for this exactly? Is there any harm in not having this check?

Reply:
Nevermind, I'm dumb. Get it now.

Reply:
So, the way you have it now, the 'list-searched' class will be added to artlist every time the loop runs, which could be wasteful depending on how JS internally does this.

The main reason I did it this way is that if you use the count then if there are no matches then it also removes the class. I guess whether or not that is what you want is something different. If you don't, then isMatch can be removed.

I'd still keep the `hasFilter` check though, because otherwise if the user backspaces the clear the search instead of clicking the clear button, the class won't get removed.